	Backport from mainline (originally from Sourcery G++ 4.4):

	gcc/
	2010-07-07  Jie Zhang  <jie@codesourcery.com>
	* genautomata.c (output_automata_list_min_issue_delay_code):
	Correctly decompress min_issue_delay.

2010-07-15  Jie Zhang  <jie@codesourcery.com>

 	Issue #8980
 
 	Backport from mainline (originally from Sourcery G++ 4.4):

=== modified file 'gcc/genautomata.c'
--- old/gcc/genautomata.c	2009-11-25 10:55:54 +0000
+++ new/gcc/genautomata.c	2010-07-30 14:21:58 +0000
@@ -7865,12 +7865,15 @@
 	{
 	  fprintf (output_file, ") / %d];\n",
 		   automaton->min_issue_delay_table_compression_factor);
-	  fprintf (output_file, "      %s = (%s >> (8 - (",
+	  fprintf (output_file, "      %s = (%s >> (8 - ((",
 		   TEMPORARY_VARIABLE_NAME, TEMPORARY_VARIABLE_NAME);
 	  output_translate_vect_name (output_file, automaton);
+	  fprintf (output_file, " [%s] + ", INTERNAL_INSN_CODE_NAME);
+	  fprintf (output_file, "%s->", CHIP_PARAMETER_NAME);
+	  output_chip_member_name (output_file, automaton);
+	  fprintf (output_file, " * %d)", automaton->insn_equiv_classes_num);
 	  fprintf
-	    (output_file, " [%s] %% %d + 1) * %d)) & %d;\n",
-	     INTERNAL_INSN_CODE_NAME,
+	    (output_file, " %% %d + 1) * %d)) & %d;\n",
 	     automaton->min_issue_delay_table_compression_factor,
 	     8 / automaton->min_issue_delay_table_compression_factor,
 	     (1 << (8 / automaton->min_issue_delay_table_compression_factor))

